java-checks/src/main/java/org/sonar/java/checks/IfElseIfStatementEndsWithElseCheck.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.IfStatementTree;
+import org.sonar.plugins.java.api.tree.StatementTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S126")
+public class IfElseIfStatementEndsWithElseCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  private JavaFileScannerContext context;
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    this.context = context;
+    scan(context.getTree());
+  }
+
+  @Override
+  public void visitIfStatement(IfStatementTree tree) {
+    StatementTree statement = tree;
+    if (tree.elseKeyword() == null) {
+      return;
+    }
+    IfStatementTree ifStmt;
+    while (statement != null) {
+      ifStmt = (IfStatementTree) statement;
+      visit(ifStmt);
+      statement = ifStmt.elseStatement();
+      if (statement == null) {
+        context.reportIssue(this, tree, "\"if ... else if\" constructs should end with \"else\" clauses.");
+      } else if (!statement.is(Tree.Kind.IF_STATEMENT)) {
+        return;
+      }
+    }
+  }
+
+  private void visit(IfStatementTree tree) {
+    scan(tree.condition());
+    scan(tree.thenStatement());
```

Comment:
because you are not visiting the `else`, it means that you are going to fail finding these cases:
```java
    if (true) {
      // do something
    } else {
      if (true) {  //  not detected
      } else if (true) {
      }
    }
```